keel-ec2-plugin/src/main/kotlin/com/netflix/spinnaker/keel/api/ec2/ClusterSpec.kt

```diff
@@ -0,0 +1,191 @@
+package com.netflix.spinnaker.keel.api.ec2
+
+import com.fasterxml.jackson.annotation.JsonCreator
+import com.fasterxml.jackson.annotation.JsonIgnore
+import com.fasterxml.jackson.annotation.JsonUnwrapped
+import com.netflix.spinnaker.keel.api.Monikered
+import com.netflix.spinnaker.keel.api.ResourceSpec
+import com.netflix.spinnaker.keel.ec2.resource.ResolvedImages
+import com.netflix.spinnaker.keel.model.Moniker
+import java.time.Duration
+
+/**
+ * Transforms a [ClusterSpec] into a concrete model of server group desired states.
+ */
+fun ClusterSpec.resolve(resolvedImages: ResolvedImages): Set<ServerGroup> {
```

Comment:
This is a "turn Keel spec into resolved desired state" function. I wanted it to be separate here rather than buried in `ClusterHandler` for ease of testing.